src/python/pants/engine/internals/scheduler.py

```diff
@@ -679,3 +624,48 @@ def get_observation_histograms(self):
 
     def record_test_observation(self, value: int) -> None:
         self._scheduler.record_test_observation(self._session, value)
+
+
+def register_rules(rule_index: RuleIndex, union_membership: UnionMembership) -> PyTasks:
```

Comment:
Lmk if there's a better place to put this. Maybe `native.py`? We won't need the `Native` singleton because CPython allows us to directly access `native_engine.so`, and we are slowly getting rid of `Externs`, but we could still use that file to collect helper functions like this?

Reply:
This should be fine... exploding access to `native_engine` out across multiple modules makes sense, because if we could (easily) split `native_engine` we would.